ui/jobs/jobs.js

```diff
@@ -3124,6 +3226,48 @@ class Bars {
           this.buffTracker.onUseAbility(m.groups.id, m.groups);
       }
     }
+    // For extremely complex BRD
+    if (this.job != 'BRD') return;
+    if (!this.dotTarget)
+      this.dotTarget = [];
+    const brdDoTs = Object.freeze([
+      EffectId.Stormbite,
+      EffectId.Windbite,
+      EffectId.CausticBite,
+      EffectId.VenomousBite,
+    ]);
+    if (type === '26') {
+      let m = log.match(kMobGainsEffectFromYouRegex);
+      if (m) {
+        if (Object.values(brdDoTs).includes(m.groups.effectId.toUpperCase()))
+          this.dotTarget.push(m.groups.targetId);
+        let f = this.mobGainEffectFromYouFuncMap[m.groups.effectId.toUpperCase()];
+        if (f)
+          f(name, m.groups);
+      }
+    } else if (type === '30') {
+      let m = log.match(kMobLosesEffectFromYouRegex);
+      if (m) {
+        if (Object.values(brdDoTs).includes(m.groups.effectId.toUpperCase())) {
+          const index = this.dotTarget.indexOf(m.groups.targetId);
+          if (index > -1)
+            this.dotTarget.splice(index, 1);
+        }
+      }
+    } else if (type === '21' || type === '22') {
+      let m = log.match(kYouUseAbilityRegex);
+      if (m) {
+        if (this.dotTarget.includes(m.groups.targetId))
+          this.lastAttackedDotTarget = m.groups.targetId;
```

Comment:
I'm not quite sure I follow the reasoning of storing the `lastAttackedDotTarget`.  It it because the repertoire timer then resets constantly and looks weird? If that's the case, can you leave a comment about what's going on here?

Alternatively, a simpler approach might just be `lastDotTarget` where you just keep track of the last thing you've put dots on. if it's already the case that multitarget is tricky.

Reply:
For `lastDotTarget`, there is a condition that when an add appear, you just cast DoT to them and turn back to your main target. And when that add die, it need more code to redirect `lastDotTarget` to your main target. `lastAttackedTarget`, `lastDotTarget` can both not be maintarget, but `lastAttackedDotTarget` must be your main target.

Reply:
Ah, I see.  That makes sense!  Could you leave a comment about this scenario?

One other question I had was when do the `dotTargets` get removed? It seems like this list will grow forever.  Maybe it'd be worth capping at N entries and also clearing whenever you're out of combat?

Reply:
ID in `dotTargets` will be removed when they lost DoT.
Also clear `dotTargets` when passing `UpdateJob`.
I think I can clear it on zone change, or remove it when removing combatant, since ID in `dotTargets` only cannot removed when removing combatant without death, mostly because of zone change. Is this be better way?

Reply:
Oh, sorry.  You're right!  I missed the code where these are removed when the dots are removed.  Thanks for also mentioning the death case.

I think it'd still be worth clearing on zone change too, but then I think there's nothing else to do here.